src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -1525,18 +1525,21 @@ public function check_read_permission( $post ) {
 			return false;
 		}
 
+		// Accounts for attachments with an inherit status on trashed and deleted parents.
+		$post_status = get_post_status( $post );
+
 		// Is the post readable?
-		if ( 'publish' === $post->post_status || current_user_can( 'read_post', $post->ID ) ) {
+		if ( 'publish' === $post_status || current_user_can( 'read_post', $post->ID ) ) {
 			return true;
 		}
 
-		$post_status_obj = get_post_status_object( $post->post_status );
+		$post_status_obj = get_post_status_object( $post_status );
 		if ( $post_status_obj && $post_status_obj->public ) {
 			return true;
 		}
 
 		// Can we read the parent if we're inheriting?
-		if ( 'inherit' === $post->post_status && $post->post_parent > 0 ) {
+		if ( 'inherit' === $post_status && $post->post_parent > 0 ) {
```

Comment:
With the changes above, I don't think this will ever pass (`$post_status` should never be `inherit`) but will need to check/write some tests.

If it's reached, that might indicate the same problem exists in `map_meta_cap()`, ie checking the post status directly rather than via the getter function.

Reply:
Interesting. Adding test coverage would be great to confirm that!

Reply:
I discovered working on another ticket that this may pass, see the changes to `get_post_status()` in PR #863.

In case I end up splitting up the PR (which is somewhat likely), in `get_post_status()` if `$post->post_parent` is set but the post ID does not exist then `get_post_status()` can return an empty string.

Reply:
Good point, after reviewing the current code for `get_post_status` the fix proposed here will not work correctly.